terraform/lambda/main.tf

```diff
@@ -0,0 +1,241 @@
+          "secretsmanager:GetSecretValue"
+        ]
+        Resource = "*"
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "logs:CreateLogGroup"
+        ]
+        Resource = "arn:aws:logs:${var.region}:${var.aws_account_id}:*"
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "logs:CreateLogStream",
+          "logs:PutLogEvents"
+        ]
+        Resource = [
+          "arn:aws:logs:${var.region}:${var.aws_account_id}:log-group:/aws/lambda/${var.domain}-${var.service_subdomain}-lambda:*"
+        ]
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "logs:CreateLogGroup",
+          "logs:CreateLogStream",
+          "logs:PutLogEvents"
+        ]
+        Resource = "*"
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "execute-api:Invoke",
+          "execute-api:ManageConnections"
+        ]
+        Resource = "arn:aws:execute-api:*:*:*"
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "s3:*",
+          "s3-object-lambda:*"
+        ]
+        Resource = "*"
+      }
+    ]
+  })
+  depends_on = [aws_iam_role.lambda_execution_role]
+}
+
+# 6. Create the Lambda function
+resource "aws_lambda_function" "tech_audit_lambda" {
+  function_name = "${var.domain}-${var.service_subdomain}-lambda"
+  package_type  = "Image"
+  image_uri     = "${var.aws_account_id}.dkr.ecr.${var.region}.amazonaws.com/${var.ecr_repository}:${var.container_ver}"
+  
+  role = aws_iam_role.lambda_execution_role.arn
+
+  memory_size = 128
+  timeout     = 30
+
+  environment {
+    variables = {
+      SOURCE_BUCKET      = var.source_bucket
+      SOURCE_KEY         = var.source_key
+      DESTINATION_BUCKET = var.destination_bucket
+      DESTINATION_KEY    = var.destination_key
+    }
+  }
+
+  depends_on = [
+    aws_iam_role_policy.lambda_ecr_policy,
+    aws_iam_role_policy.lambda_s3_access,
+    aws_iam_role_policy.lambda_additional_permissions,
+    aws_iam_role_policy_attachment.lambda_basic_execution
+  ]
+}
+
+# 7. Add ECR policy after the lambda function is created
+resource "aws_ecr_repository_policy" "lambda_ecr_access" {
+  repository = var.ecr_repository_name
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Sid    = "LambdaECRAccess"
+        Effect = "Allow"
+        Principal = {
+          Service = "lambda.amazonaws.com"
+          AWS = aws_iam_role.lambda_execution_role.arn
+        }
+        Action = [
+          "ecr:BatchGetImage",
+          "ecr:GetDownloadUrlForLayer"
+        ]
+      }
+    ]
+  })
+
+  depends_on = [aws_iam_role.lambda_execution_role]
+}
+
+# Add EventBridge (CloudWatch Events) rule for daily trigger
+resource "aws_cloudwatch_event_rule" "daily_trigger" {
+  name                = "${var.domain}-${var.service_subdomain}-daily-trigger"
+  description         = "Triggers tech radar lambda daily"
+  schedule_expression = "cron(0 10 * * ? *)"
```

Comment:
More of an FYI, Lambda can also use `rate()`. So for daily execution, you can do `rate(1 day)`. More on this [here](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-scheduled-rule-pattern.html#eb-rate-expressions) if you're interested.

Reply:
Yeah, I just though I'd set it to run once a day at 10am.